src/main/Extensions/VSCode/VSCodeExtension.ts

```diff
@@ -44,20 +46,25 @@ export class VSCodeExtension implements Extension {
     public constructor(
         private readonly operatingSystem: OperatingSystem,
         private readonly assetPathResolver: AssetPathResolver,
+        private readonly logger: Logger,
         private readonly settingsManager: SettingsManager,
         private readonly fileImageGenerator: FileImageGenerator,
     ) {}
 
     async getSearchResultItems(): Promise<SearchResultItem[]> {
-        const recentPaths = this.getRecents();
+        const searchResults: SearchResultItem[] = [];
 
-        const searchItems = await Promise.all(
-            recentPaths.map((recent) => {
-                return this.getSearchItem(recent);
-            }),
-        );
+        for (const recent of this.getRecents()) {
+            try {
+                searchResults.push(await this.getSearchItem(recent));
+            } catch (error) {
+                const uri = this.getUri(recent);
+                this.logger.error("VSCODE: " + uri + " " + error);
```

Comment:
Please use the extension id here:

```suggestion
                this.logger.error(this.id + ": " + uri + " " + error);
```